app/com/linkedin/drelephant/ElephantRunner.java

```diff
@@ -80,13 +95,17 @@ public void run() {
         @Override
         public Void run() {
           HDFSContext.load();
+          loadGeneralConfiguration();
           loadAnalyticJobGenerator();
           ElephantContext.init();
 
-          _service = Executors.newFixedThreadPool(EXECUTOR_NUM);
           _jobQueue = new LinkedBlockingQueue<AnalyticJob>();
-          for (int i = 0; i < EXECUTOR_NUM; i++) {
-            _service.submit(new ExecutorThread(i + 1, _jobQueue));
+          logger.info("executor num is " + _executorNum);
+          if (_executorNum > 0) {
+            _service = Executors.newFixedThreadPool(EXECUTOR_NUM);
```

Comment:
Why do you still make use of the hardcoded value here? Should it not be '_executorNum'?